sdk/python/kfp/components/_data_passing.py

```diff
@@ -35,6 +35,36 @@
 ])
 
 
+def _serialize_str(str_value: str) -> str:
+    if not isinstance(str_value, str):
+        raise TypeError('Value "{}" has type "{}" instead of str.'.format(str(str_value), str(type(str_value))))
+    return str_value
+
+
+def _serialize_int(int_value: int) -> str:
+    if isinstance(int_value, str):
+        return int_value
+    if not isinstance(int_value, int):
```

Comment:
I think you can combine this two if by saying `if not isinstance(int_value, int) and not isinstance(int_value, str)`?

Reply:
Similar for float serializer

Reply:
This will work in this case, but it might not work for general serializer structure:

```python
def serialize_some_type(value: SomeType):
    if isinstance(value, str):
        return value
    if not isinstance(value, SomeType):
        raise TypeError('Value "{}" has type "{}" instead of SomeType.'.format(str(value), str(type(value))))
    return serialize(value)
```

Reply:
I find the existing code structure a little bit more readable with three clear cases.